frontera/utils/misc.py

```diff
@@ -29,11 +32,11 @@ def load_object(path):
 
 
 def get_crc32(name):
-    return crc32(name) if type(name) is str else crc32(name.encode('utf-8', 'ignore'))
+    return crc32(to_bytes(name, 'utf-8', 'ignore'))
 
 
 def chunks(l, n):
-    for i in xrange(0, len(l), n):
+    for i in range(0, len(l), n):
```

Comment:
There's also the [grouper recipe from itertools](https://docs.python.org/3/library/itertools.html#itertools-recipes)
```
def grouper(iterable, n, fillvalue=None):
    "Collect data into fixed-length chunks or blocks"
    # grouper('ABCDEFG', 3, 'x') --> ABC DEF Gxx"
    args = [iter(iterable)] * n
    return zip_longest(*args, fillvalue=fillvalue)
```